acsoo/tools.py

```diff
@@ -26,6 +28,50 @@ def _escape(s):
     return s
 
 
+@contextmanager
+def tempinput(data):
+    temp = tempfile.NamedTemporaryFile(delete=False)
+    temp.write(data)
+    temp.close()
+    try:
+        yield temp.name
+    finally:
+        os.unlink(temp.name)
+
+
+def cmd_check_current_branch(branches):
+    current_branch = check_output(
+        ['git', 'rev-parse', '--abbrev-ref', 'HEAD'])
+    current_branch = current_branch.replace("\n", "")
+    if current_branch not in branches:
+        return False
+    else:
+        return True
+
+
+def cmd_push(paths_to_push, message, skip_ci=True):
+    out = check_output(['git', 'ls-files', '--other', '--exclude-standard',
+                        '--modified'])
+    list_out = out.splitlines()
+    items_to_remove = set([])
+    for item in paths_to_push:
+        if item not in list_out:
+            click.echo("No change on %s" % item)
+            items_to_remove.add(item)
```

Comment:
`git add` on an unmodified file does nothing, so maybe it's not necessary to do the git ls-files part. You could simply git 
 `git add` all files and at the end do a `git diff --cached --exit-code` to see if there is something to commit.